DataStructures/Source/LinkedList/LinkedList.cs

```diff
@@ -47,41 +47,42 @@ public void AddAt(int index, T item)
                 return;
             }
 
-            var previous = ElementAt(index - 1, _head);
+            var previous = FindNodeAt(index - 1);
             previous.Next = new Node<T>(item, previous.Next);
             Length++;
         }
 
         public void Remove(T item)
         {
-            var previous = _head;
+            Node<T> previous = null;
```

Comment:
If I get you correctly, in case they are defined inside a for-loop there will be more code and it wouldn't be that readable + the compiler gets angry.
![image](https://cloud.githubusercontent.com/assets/9809523/26445973/da2fd402-414a-11e7-9cf2-715d710d7104.png)